packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js

```diff
@@ -208,8 +222,28 @@ export default ({ coreSagas }) => {
 
   const cancelISX = function * () {
     const modals = yield select(selectors.modals.getModals)
-    if (path(['type'], head(modals)) === 'CoinifyExchangeData') yield put(actions.modals.closeAllModals())
-    else yield put(A.coinifyNextCheckoutStep('checkout'))
+    const trade = yield select(selectors.core.data.coinify.getTrade)
+
+    if (path(['type'], head(modals)) === 'CoinifyExchangeData') {
+      yield put(actions.modals.closeAllModals())
+    } else if (trade.data.state === 'awaiting_transfer_in') {
+      yield put(actions.form.change('buySellTabStatus', 'status', 'order_history'))
+      yield put(A.coinifyNextCheckoutStep('checkout'))
+    } else {
+      yield put(A.coinifyNextCheckoutStep('checkout'))
+    }
+  }
+
+  const cancelTrade = function * (data) {
+    const trade = data.payload
+    try {
+      yield put(A.setCancelTradeId(trade.id))
+      yield put(A.coinifyLoading())
+      yield call(coreSagas.data.coinify.cancelTrade, { trade })
+      yield put(A.coinifySuccess())
```

Comment:
Personally I think it makes more sense to cancelTradeLoading() -> cancelTradeSuccess() || cancelTradeFailure(), and set busy based on that cata. What if another component sets coinifyLoading somewhere else? Now cancelTrade is loading too.